primitives/account/src/lib.rs

```diff
@@ -73,6 +73,14 @@ impl Into<[u8; 20]> for AccountId20 {
 	}
 }
 
+impl From<[u8; 32]> for AccountId20 {
+	fn from(bytes: [u8; 32]) -> Self {
+		let mut buffer = [0u8; 20];
+		buffer.copy_from_slice(&bytes[..20]);
+		Self(buffer)
+	}
+}
```

Comment:
Overall you are right, but this conversion is only made because the conversion built by polkadot works with 32 byte accounts, so we need to pass from there to a 20 byte account.

This has no security concerns because the 32 byte account is derived with a hash in the first place.

Reply:
The particular usage of it isn't what concerns me, it's having it in the first place. Couldn't we create a fn that needs to be explicitly called, something like `unsafe_from_bytes_32` or `from_known_secure_bytes` or something like that? Doing so would obviously avoid any possible confusion.

Reply:
Or is there a requirement that there be a `From` impl for it to work?

Reply:
@notlesh the `From<[u8; 32]>` impl is a requirement from `ForeignChainAliasAccount`, it can't compile without it